### PR TITLE
Feature/pm 103 imu to urdf

### PR DESCRIPTION
**Pull request by RutgerVanBeek**
Depends on march/hardware-interface#171

Closes PM-103

## Description

This links an Xsens IMU in the backpack to the URDF. This means that the orientation can be calculated for all the parts in the URDF. This way the orientation can be visualized in Rviz and the center of mass can be calculated with respect to a point outside of the exoskeleton. 
## Changes

In the march data collector there is a callback function for the IMU messages which creates a transform from the world to the exoskeleton. 
In the URDF I added an imu_link which is what ROS recommend for sensors. 
When there is no IMU data available the robot is connected to the world via the imu_link

### CODE REVIEW

File: march_data_collector/src/march_data_collector/march_data_collector_node.py [hunk 1]
```diff
@@ -17,9 +21,33 @@ def ImcStateCallback(data):
     rospy.logdebug('received IMC message current is ' + str(data.current))
 
 
+def IMUCallback(data, IMUbroadcaster):
+    if data.header.frame_id == "imu_link":
```
Comment (Olavhaasie) [inline march_data_collector/src/march_data_collector/march_data_collector_node.py]: Please use single quotes everywhere
```suggestion
    if data.header.frame_id == 'imu_link':
```

File: march_data_collector/src/march_data_collector/march_data_collector_node.py [hunk 2]
```diff
@@ -17,9 +21,33 @@ def ImcStateCallback(data):
...
+        transform = TransformStamped()
+
+        transform.header.stamp = rospy.Time.now()
+        transform.header.frame_id = "world"
+        transform.child_frame_id = "imu_link"
```
Comment (Olavhaasie) [inline march_data_collector/src/march_data_collector/march_data_collector_node.py]:
```suggestion
        transform.child_frame_id = 'imu_link'
```

File: march_data_collector/src/march_data_collector/march_data_collector_node.py [hunk 3]
```diff
@@ -17,9 +21,33 @@ def ImcStateCallback(data):
...
+        transform.transform.rotation.w = imu_rotation[3]
+
+        IMUbroadcaster.sendTransform(transform)
+
+
 def main():
     rospy.init_node('data_collector', anonymous=True)
     joint_names = rospy.get_param('/march/joint_names')
+
+    IMUbroadcaster = tf2_ros.TransformBroadcaster()
```
Comment (Olavhaasie) [inline march_data_collector/src/march_data_collector/march_data_collector_node.py]:
```suggestion
    imu_broadcaster = tf2_ros.TransformBroadcaster()
```

File: march_data_collector/src/march_data_collector/march_data_collector_node.py [hunk 4]
```diff
@@ -28,4 +56,6 @@ def main():
 
     IMCStateSubscriber = rospy.Subscriber('/march/imc_states', ImcErrorState, ImcStateCallback)
 
+    IMUSubscriber = rospy.Subscriber('/march/imu', Imu, IMUCallback, (IMUbroadcaster))
```
Comment (Olavhaasie) [inline march_data_collector/src/march_data_collector/march_data_collector_node.py]:
```suggestion
    IMUSubscriber = rospy.Subscriber('/march/imu', Imu, imu_callback, imu_broadcaster)
```

File: march_description/urdf/march-iv.xacro [hunk 5]
```diff
@@ -472,8 +476,13 @@
 
 
     <!-- Joints -->
-    <joint name="hip" type="fixed">
+    <joint name="imu_joint" type="fixed">
         <parent link="${parent_name}"/>
+        <child link="imu_link"/>
+    </joint>
```
Comment (Olavhaasie) [inline march_description/urdf/march-iv.xacro]: Is still necessary since a link is also added to the `hip`?

**Comment by Olavhaasie** [comment]
> It would be nice if you could also adjust the naming of the other functions innocent : https://www.python.org/dev/peps/pep-0008/

In CLion you can set to give warnings or errors on these violations. Under `Settings > Editor > Inspections` and then searchin for pep8. You should set those to warning or error.

File: march_description/urdf/march-iv.xacro (hunk 5 above)
Comment (RutgerVanBeek) [reply march_description/urdf/march-iv.xacro]: This connects the robot to the world. parent_name is set to world.

Comment (RutgerVanBeek) [reply march_description/urdf/march-iv.xacro]: This connects the imu_link to the world.

File: march_data_collector/src/march_data_collector/march_data_collector_node.py (hunk 3 above)
Comment (Ishadijcks) [reply march_data_collector/src/march_data_collector/march_data_collector_node.py]: Can't your linter pick this up?

**Comment by Olavhaasie** [comment]
@RutgerVanBeek You have merge conficts

**Pull request merged**